### PR TITLE
Fail to launch api unit test from arbitrary path

### PR DESCRIPTION
**Pull request by cdlliuy**
tracker [#154910357]

### CODE REVIEW

File: api/lib/utils/metricsCollectorUtils.js [hunk 1]
```diff
@@ -14,6 +14,18 @@ module.exports = function(metricsCollectorSettings) {
...
+        throw new Error("Invalid TLS key path: " + metricsCollectorSettings.tls.keyFile);
+      }
+      if(!fs.existsSync(metricsCollectorSettings.tls.certFile)){
+        logger.error("Invalid TLS certificate path: " + metricsCollectorSettings.tls.certFile);
+        throw new Error("Invalid TLS certificate path: " + metricsCollectorSettings.tls.certFile);
+      }
+      if(!fs.existsSync(metricsCollectorSettings.tls.caCertFile)){
+        logger.error("Invalid TLS ca certificate path: " + metricsCollectorSettings.tls.caCertFile);
+        throw new Error("Invalid TLS ca certificate path: " + metricsCollectorSettings.tls.caCertFile);
+      }
```
Comment (qibobo) [inline api/lib/utils/metricsCollectorUtils.js]: where is the relevant test case?

File: api/lib/utils/scalingEngineUtils.js [hunk 2]
```diff
@@ -14,6 +14,18 @@ module.exports = function(scalingEngineSettings) {
...
+        throw new Error("Invalid TLS key path: " + scalingEngineSettings.tls.keyFile);
+      }
+      if(!fs.existsSync(scalingEngineSettings.tls.certFile)){
+        logger.error("Invalid TLS certificate path: " + scalingEngineSettings.tls.certFile);
+        throw new Error("Invalid TLS certificate path: " + scalingEngineSettings.tls.certFile);
+      }
+      if(!fs.existsSync(scalingEngineSettings.tls.caCertFile)){
+        logger.error("Invalid TLS ca certificate path: " + scalingEngineSettings.tls.caCertFile);
+        throw new Error("Invalid TLS ca certificate path: " + scalingEngineSettings.tls.caCertFile);
+      }
```
Comment (qibobo) [inline api/lib/utils/scalingEngineUtils.js]: same as above

File: api/lib/utils/schedulerUtils.js [hunk 3]
```diff
@@ -19,6 +19,18 @@ module.exports = function(schedulerSettings) {
...
+        throw new Error("Invalid TLS key path: " + schedulerSettings.tls.keyFile);
+      }
+      if(!fs.existsSync(schedulerSettings.tls.certFile)){
+        logger.error("Invalid TLS certificate path: " + schedulerSettings.tls.certFile);
+        throw new Error("Invalid TLS certificate path: " + schedulerSettings.tls.certFile);
+      }
+      if(!fs.existsSync(schedulerSettings.tls.caCertFile)){
+        logger.error("Invalid TLS ca certificate path: " + schedulerSettings.tls.caCertFile);
+        throw new Error("Invalid TLS ca certificate path: " + schedulerSettings.tls.caCertFile);
+      }
```
Comment (qibobo) [inline api/lib/utils/schedulerUtils.js]: same as above

**Comment by boyang9527** [comment]
why we need to run unit test in any directory?

**Comment by cdlliuy** [comment]
@boyang9527, I found the issue when adding more testcases for api validation.   The test configuration here is wrong,  and it is a broken experience in unit test.

File: api/lib/utils/scalingEngineUtils.js (hunk 2 above)
Comment (cdlliuy) [reply api/lib/utils/scalingEngineUtils.js]: I removed this from this PR. will resubmit later

File: api/lib/utils/metricsCollectorUtils.js (hunk 1 above)
Comment (cdlliuy) [reply api/lib/utils/metricsCollectorUtils.js]: I removed this from this PR. will resubmit later

**Comment by qibobo** [comment]
LGTM

**Pull request merged**